tests/icons.test.js

```diff
@@ -5,28 +5,54 @@ icons.forEach(icon => {
   const filename = getIconSlug(icon);
   const subject = require(`../icons/${filename}.js`);
 
-  test(`${icon.title} has a "title"`, () => {
+  test(`${icon.title} has the correct "title"`, () => {
     expect(typeof subject.title).toBe('string');
+    expect(subject.title).toEqual(icon.title);
   });
 
-  test(`${icon.title} has a "hex" value`, () => {
+  test(`${icon.title} has the correct "slug"`, () => {
+    expect(typeof subject.slug).toBe('string');
+    expect(subject.slug).toEqual(getIconSlug(icon));
+  });
+
+  test(`${icon.title} has the correct "hex" value`, () => {
     expect(typeof subject.hex).toBe('string');
-    expect(subject.hex).toHaveLength(6);
+    expect(subject.hex).toEqual(icon.hex);
   });
 
-  test(`${icon.title} has a "source"`, () => {
+  test(`${icon.title} has the correct "source"`, () => {
     expect(typeof subject.source).toBe('string');
+    expect(subject.source).toEqual(icon.source);
   });
 
-  test(`${icon.title} has an "svg"`, () => {
+  test(`${icon.title} has an "svg" value`, () => {
     expect(typeof subject.svg).toBe('string');
   });
 
-  test(`${icon.title} has a "path"`, () => {
+  test(`${icon.title} has a valid "path" value`, () => {
     expect(typeof subject.path).toBe('string');
+    expect(subject.path).toMatch(/^[MmZzLlHhVvCcSsQqTtAaEe0-9-,.\s]+$/g);
   });
 
-  test(`${icon.title} has a "slug"`, () => {
-    expect(typeof subject.slug).toBe('string');
+  test(`${icon.title} has ${icon.guidelines ? "the correct" : "no"} "guidelines"`, () => {
+    if (icon.guidelines) {
+      expect(typeof subject.guidelines).toBe('string');
+    } else {
+      expect(subject.guidelines).toBeUndefined();
+    }
+  });
+
+  test(`${icon.title} has ${icon.license ? "the correct" : "no"} "license"`, () => {
+    if (icon.license) {
+      expect(typeof subject.license).toBe('object');
+      expect(subject.license).toHaveProperty('type', icon.license.type);
+      if (icon.license.type === "custom") {
+        expect(subject.license).toHaveProperty('url', icon.license.url);
+      } else {
+        expect(typeof subject.license.url).toBe('string');
```

Comment:
Why not adding equality comparison for predetermined SPDX license urls?

```suggestion
        expect(typeof subject.license.url).toBe('string');
        expect(subject.license.url).toEqual(`https://spdx.org/licenses/${icon.license.type}.html`);
```

Reply:
I'm fine adding it. I initially didn't because it is literally copying the source we're trying to test. That said it's not a bad idea to test it for regression.

A compromise could be to test if it is _a_ URL (i.e. matches a URL pattern) instead. This way we don't need to update the test if we/SPDX decides to change these URLs.

What do you think @mondeja?

Reply:
I agree, a regex would also avoid regressions here, so as you prefer @ericcornelissen

Reply:
Added this in 9e71e9674dd45c328d2284fedf7ae9c32708d0b2 :+1: Should be all good now